### PR TITLE
[MINOR][SQL] Add disable bucketedRead workaround when throw RuntimeException

### PR DESCRIPTION
**Pull request by wangyum**
## What changes were proposed in this pull request?
It will throw `RuntimeException` when read from bucketed table(about 1.7G per bucket file):
image

Default(enable bucket read):
image

Disable bucket read:
image

The reason is that each bucket file is too big. a workaround is disable bucket read. This PR add this workaround to Spark.

## How was this patch tested?

manual tests

### CODE REVIEW

File: sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/WritableColumnVector.java [hunk 1]
```diff
@@ -101,7 +101,8 @@ private void throwUnsupportedException(int requiredCapacity, Throwable cause) {
     String message = "Cannot reserve additional contiguous bytes in the vectorized reader (" +
         (requiredCapacity >= 0 ? "requested " + requiredCapacity + " bytes" : "integer overflow") +
         "). As a workaround, you can reduce the vectorized reader batch size, or disable the " +
-        "vectorized reader. For parquet file format, refer to " +
+        "vectorized reader, or disable " + SQLConf.BUCKETING_ENABLED().key() + " if you read " +
+        "from bucket table. For parquet file format, refer to " +
```
Comment (HyukjinKwon) [inline sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/WritableColumnVector.java]: parquet -> Parquet

**Comment by HyukjinKwon** [comment]
> The reason is that each bucket file is too big

Can you elaborate please? Is it because we don't chunk each file into multiple splits when we read bucketed table?

**Review by HyukjinKwon — APPROVED** [review]

**Comment by wangyum** [comment]
retest this please

**Comment by wangyum** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**